ovs/lib/vdisk.py

```diff
@@ -526,26 +527,37 @@ def create_new(diskname, size, storagedriver_guid):
 
     @staticmethod
     @celery.task(name='ovs.vdisk.create_volume')
-    def create_volume(location, size):
+    def create_volume(location, size, storagerouter_guid=None):
         """
         Create a volume using filesystem calls
         Calls "truncate" to create sparse raw file
-        TODO: use volumedriver API
-        TODO: model VDisk() and return guid
 
         @param location: location, filename
         @param size: size of volume, GB
+        @param: storagerouter_guid: use ssh client to create file on remote storagerouter
         @return None
         """
         logger.info('Creating volume {0} of {1} GB'.format(location, size))
-        if os.path.exists(location):
-            raise RuntimeError('File already exists at %s' % location)
+        if storagerouter_guid is None:
+            if os.path.exists(location):
+                raise RuntimeError('File already exists at %s' % location)
 
-        output = check_output('truncate -s {0}G "{1}"'.format(size, location), shell=True).strip()
-        output = output.replace('\xe2\x80\x98', '"').replace('\xe2\x80\x99', '"')
+            output = check_output('truncate -s {0}G "{1}"'.format(size, location), shell=True).strip()
+            output = output.replace('\xe2\x80\x98', '"').replace('\xe2\x80\x99', '"')
 
-        if not os.path.exists(location):
-            raise RuntimeError('Cannot create file %s. Output: %s' % (location, output))
+            if not os.path.exists(location):
+                raise RuntimeError('Cannot create file %s. Output: %s' % (location, output))
+        else:
+            storagerouter = StorageRouter(storagerouter_guid)
+            client = SSHClient(storagerouter.ip)
```

Comment:
No issue, but just a tip
You can pass a storagerouter object to the SSHClient, no IP required